src/main/java/org/c02e/jpgpj/Encryptor.java

```diff
@@ -408,13 +411,68 @@ public void encrypt(InputStream plaintext, OutputStream ciphertext)
      * @throws PassphraseException if an incorrect passphrase was supplied
      * for one of the signing keys.
      */
-    public void encrypt(InputStream plaintext, OutputStream ciphertext,
-    FileMetadata meta) throws IOException, PGPException {
+    public void encrypt(
+        InputStream plaintext, OutputStream ciphertext, FileMetadata meta)
+            throws IOException, PGPException {
+        if (meta == null) {
+            meta = new FileMetadata();
+        }
+
+        try (OutputStream targetStream = prepareCiphertextOutputStream(ciphertext, meta)) {
+            // copy plaintext bytes into encryption pipeline
+            copy(plaintext, targetStream, null /* signer is inside the target */, meta);
+        }
+    }
+
+    public OutputStream prepareCiphertextOutputStream(File ciphertext)
+            throws IOException, PGPException {
+        // delete old output file
+        if (ciphertext.delete()) {
+            log.debug("prepareCiphertextOutputStream - deleted {}", ciphertext);
+        }
+
+        FileMetadata meta = new FileMetadata(ciphertext);
+        OutputStream fileStream = null;
+        try {
+            fileStream = new FileOutputStream(ciphertext);
+            OutputStream wrapper = prepareCiphertextOutputStream(fileStream, meta);
+            fileStream = null;  // avoid auto-close at finally clause
+            return wrapper;
+        } catch(Exception e) {
+            // delete output file if anything went wrong
+            if (fileStream != null) {
+                ciphertext.delete();
+            }
+            throw e;
+        } finally {
+            if (fileStream != null) {
+                fileStream.close();
```

Comment:
is there a case where we don't execute the catch block but still want to close the file in the finally block? -- otherwise seems like this line could be moved into the catch block (and the finally block omitted)

Reply:
We always want to close the file stream (if not null) so in the spirit of D.R.Y. (Don't Repeat Yourself) feels that the best place for it is in the _finally_ clause.